watertap/unit_models/ion_exchange_0D.py

```diff
@@ -0,0 +1,1352 @@
+            doc="Pressure drop eq C",
+        )
+
+        # ====== Resin variables ====== #
+
+        self.resin_max_capacity = Var(
+            initialize=5,
+            units=pyunits.mol / pyunits.kg,
+            bounds=(0.5, 10),  # Perry's
+            doc="Resin max capacity [mol/kg]",
+        )
+
+        self.resin_eq_capacity = Var(
+            initialize=1,
+            units=pyunits.mol / pyunits.kg,
+            bounds=(0.5, 9),  # Perry's
+            doc="Resin equilibrium capacity [mol/kg]",
+        )
+
+        self.resin_diam = Var(
+            initialize=9e-4,
+            bounds=(5e-4, 1.5e-3),
+            units=pyunits.m,  # Perry's
+            doc="Resin bead diameter [m]",
+        )
+
+        self.resin_bulk_dens = Var(
+            initialize=0.7,
+            bounds=(0.65, 0.95),  # Perry's
+            units=pyunits.kg / pyunits.L,
+            doc="Resin bulk density [kg/L]",
+        )
+
+        self.resin_particle_dens = Var(
+            initialize=1.4,
+            bounds=(0.5, None),
+            units=pyunits.kg / pyunits.L,
+            doc="Resin particle density [kg/L]",
+        )
+
+        self.K_eq = Var(
+            ion_set,
+            initialize=1.5,
+            bounds=(0.5, 10),
+            units=pyunits.dimensionless,
+            doc="Selectivity coefficient [-]",
+        )
+
+        self.R_eq = Var(
+            ion_set,
+            initialize=0.2,
+            bounds=(0, 1.1),
+            units=pyunits.dimensionless,
+            doc="Separation factor [-]",
+        )
+
+        self.resin_surf_per_vol = Var(
+            initialize=3333.33,
+            bounds=(0, 1e5),
+            units=pyunits.m**-1,
+            doc="Resin surface area per volume [m-1]",
+        )
+
+        # ====== Bed/Column variables ====== #
+
+        self.bed_vol = Var(
+            initialize=2,
+            bounds=(0.1, 75),
+            units=pyunits.m**3,
+            doc="Bed volume of one unit [m3]",
+        )
+
+        self.bed_diam = Var(
+            initialize=0.4,
+            bounds=(0.01, 4),  # DOW
+            units=pyunits.m,
+            doc="Bed diameter [m]",
+        )
+
+        self.bed_depth = Var(
+            initialize=4, bounds=(0.1, 12), units=pyunits.m, doc="Bed depth [m]"
+        )
+
+        self.bed_area = Var(
+            initialize=0.63,
+            units=pyunits.m**2,
+            doc="Bed area [m2]",
+        )
+
+        self.bed_porosity = Var(
+            initialize=0.5,
+            bounds=(0.45, 0.65),
+            units=pyunits.dimensionless,
+            doc="Bed porosity [-]",
+        )
+
+        self.col_height = Var(
+            initialize=2,
+            bounds=(0.6, 12),
+            units=pyunits.m,
+            doc="Column height [m]",
+        )
+
+        self.col_vol = Var(
+            initialize=10,
+            units=pyunits.m**3,
+            doc="Column volume of one unit [m3]",
+        )
+
```

Comment:
Some of the  vars carry a relatively small bound , which would be nice for the solver.  But I wonder if there will be any practical situation where a given bound is too narrow.  So maybe consider enlarging some bounds without hurting the solver's performance.

Reply:
The bounds are more about the costing data I have, less about the solver.

Reply:
My suggestion was considering making the model capable of solving problems in larger var domains (if they are practical), which will translate to the problem's constraints when it is solved.